### PR TITLE
Fix issues with using absolute paths for modules

### PR DESCRIPTION
**Pull request by torfason**
It seems `file.path()` is not as smart as I thought about absolute paths. When given an absolute path relative to ".", it converts it into an unusable relative path. I added a new function, `file_path()`, which returns absolute paths unchanged. 

Credit to Henrik Bengtson and the `R.utils` package for the `isAbsolutePath()` function which does some heavy lifting to determine on multiple OSs whether a path is absolute. I copied the function rather than importing the whole package, which felt like overkill.

This also allowed me to add tests for downloading URLs using the `pins` package (the local file name returned by `pins::pin()` is an absolute file name.

### CODE REVIEW

File: R/file_path.R [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#' Return a file path given file name and directory
+#'
+#' Given a directory and a file name (path), return the combined path.
+#' For relative file names, this is identical to \code{file.path()},
+#' for absolute file names, it simply returns the absolute file nema
```
Comment (smbache) [inline R/file_path.R:5]:
```suggestion
#' for absolute file names, it simply returns the absolute file name
```

File: R/file_path.R [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+#'
+#' @param working_directory character: the working directory to use
+#' @param file_name character: the name of a possible R script file.
+#' @return logical
```
Comment (smbache) [inline R/file_path.R:9]: I assume the return here is not logical :)

**Comment by smbache** [comment]
interestingly the CI doesn't run here - I think if you add `pull_request` to the `on` array in the actions yml, it may trigger correctly?

**Comment by smbache** [comment]
can you give an example of where file.path is unpredictable? is it a windows thing?

**Comment by torfason** [comment]
``` r
file.path(".","/dev/null")
#> [1] ".//dev/null"
import:::file_path(".","/dev/null")
#> [1] "/dev/null"
```

Not so much that `file.path()`is unpredictable, just that I didn't predict it. It converts all absolute paths into relative paths when prefixed with "." to implement the `.directory` parameter, I thought that it was cleanest to use "." as a default, requiring the use of a`file.path()`variant. But that caused issues with absolute paths, and figuring out whether a path is absolute turns out to be nontrivial.

Now that I think about it, it might be possible to simply rewrite `file_path()` to return the original file name _whenever the first argument is "."_ instead of whenever the second argument is an absolute path. If that works, that would alleviate the need to pull in this huge function ...

**Comment by torfason** [comment]
If I redo the pull request fresh from master, a side effect might be to have the tests pass. It might be something about how the  CI confiq was merged into the pull request after it was created ...

**Comment by smbache** [comment]
Is it cleaner to just use getwd() instead of using "."?

Re the CI, I think if you change 
```
on: [push]
```
to
```
on: [push, pull_request]
```
it should pick it up.. just a guess :) no harm trying.

**Comment by torfason** [comment]
I'm afraid getwd() wont help:

``` r
# file.path()
file.path(".","/dev/null")
#> [1] ".//dev/null"
file.path(getwd(),"/dev/null")
#> [1] "/Users/torfason/demodir//dev/null"

# Contrast with import::file_path()
import:::file_path(".","/dev/null")
#> [1] "/dev/null"
import:::file_path(getwd(),"/dev/null")
#> [1] "/dev/null"
```

And on closer thought I'm not to fond of special handling of ".". What if someone has most of his helpers in `~/utils', but then needs to pass one helper though absolute path:

```
import::from(util.R, things, .directory="~/utils")
import::from("/shared/util.R", shared_things, .directory="~/utils") # error (this could be inside a function, using character_only
```

It seems to be an effort to explain to that person that they need to delete `.directory`or set to "." to trigger some special handling (I guess they could set to "/" for absolute files, but that would require them do do the heavy lifting of figuring out if the script is an absolute path.

So I'm back to thinking `isAbsolutePat()` from `R.utils` is the best option, even if it is big it is presumably decently tested and passes all tests on all platforms (windows as well, it is just the CI that has issues).

I'll fix other issues, including the attempted fix for CI, and update the PR.

**Comment by torfason** [comment]
@smbache, it would be great to get your thoughts on whether you think it's OK to merge this pull request into master (I can then perform the actual merge). Would be good to clear this out of the way before preparing a PR to fix the recursive call issues.

**Review by smbache — APPROVED** [review]

**Pull request merged**